collection/collection.go

```diff
@@ -0,0 +1,78 @@
+package collection
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/qri-io/qri/collection/list"
+	"github.com/qri-io/qri/event"
+	"github.com/qri-io/qri/profile"
+)
+
+// RootUserKeyID is a special key identifer that lists *all* datasets in a
+// collection
+const RootUserKeyID = profile.ID("root")
+
+type Collection interface {
+	List(ctx context.Context, pid profile.ID, lp list.Params) ([]Item, error)
```

Comment:
I'm very happy with this interface and it flows nicely with the hosted version and the logic of the previews.